format/amf3/Tools.hx

```diff
@@ -56,9 +58,15 @@ class Tools {
 			case cast Array:
 				var o : Array<Dynamic> = o;
 				var a = new Array();
-				for(v in o)
-					a.push(encode(v));
+				for(k in Reflect.fields(o))
```

Comment:
I would like to use Haxe's feature to add dynamic fields to emulate the associate array values you get in amf3. I suppose we could change this to return an AssociativeArray or Amf3Array `{ m:Map<String,Dynamic>, a:Array }` instead?